### DISCUSSION TITLE
Custom action

### DISCUSSION THREAD

**Discussion by yula-21** [Security-Onion-Solutions/securityonion#7032] [Unsupported Versions] [answered]
Hi,
I was wondering if theres an option to create a custom action in the alerts page to run a script.

I know that theres a way to add custom actions in the menu actions json file but the only examples i saw were of an action sending you to a web page.
I am working on an airgap environment so my question is can I make the action run a script as I choose it instead of directing me to the internet.

**Comment by dougburks** [comment, marked as answer]
Take a look at the background actions description and see if that would work for you:
https://docs.securityonion.net/en/2.3/alerts.html#actions

**Reply by yula-21** [reply]
I have seen the documentation and the concept of background actions does fit what I want to do.

The problem is its not very well explained in the documentation, that is the reason I started the discussion in the first place.

**Reply by dougburks** [reply]
That section has moved to a new page:
https://docs.securityonion.net/en/2.3/soc-customization.html#action-menu

> The problem is its not very well explained in the documentation, that is the reason I started the discussion in the first place.

Perhaps if you could help us understand what exactly is not very well explained then we could improve it. Alternatively, you could always submit Pull Requests to improve the documentation directly.

**Reply by yula-21** [reply]
The fields are not explained,
I don’t understand what to put in each field (in the background actions as well as the regular actions)
Can you please explain the syntax?

**Reply by dougburks** [reply]
Let's start with a regular action. Here's the example from the documentation.

For example, suppose you want to add AbuseIPDB with URL https://www.abuseipdb.com/check/{value}. First, copy /opt/so/saltstack/default/salt/soc/files/soc/menu.actions.json to /opt/so/saltstack/local/salt/soc/files/soc/menu.actions.json. Next, edit /opt/so/saltstack/local/salt/soc/files/soc/menu.actions.json using your favorite text editor and insert the following as the next to last line of the file:
```
,{ "name": "AbuseIPDB", "description": "Search for this value at AbuseIPDB", "icon": "fa-external-link-alt", "target": "_blank","links": [ "https://www.abuseipdb.com/check/{value}" ]}
```

So in this example:
- `AbuseIPDB` is what's going to display on the Actions menu.
- `Search for this value at AbuseIPDB` is what's going to display when you hover over the `AbuseIPDB` option on the Actions menu.
- `https://www.abuseipdb.com/check/{value}` is where the browser will go when you click the `AbuseIPDB` option on the Actions menu.

Is this what you're looking for?
